### ISSUE TITLE
Potentially investigate and tweaks some util functions

### ISSUE THREAD

**Issue by pattonwebz** [equalizedigital/accessibility-checker#1150]
When reviewing some session data from an agent I had asked to write some tests I noticed it was referencing things it was confused by in the source functions. I asked it to analyse and explain and this is what it came up with. Some of this makes sense on initial read but should be validate properly.

> when you were working on this last I noticed in your session output you mentioned a number of times that things were working in unexpected ways. That may indicate that the actual code under t...

I analyzed the test suite and found several quirks and edge cases that indicate potential issues with the underlying functions. Here's a summary of the concerning behaviors:

### 1. `edac_days_active()` - Backwards Date Calculation
**Issue**: The function calculates `activation_date - current_date` instead of the intuitive `current_date - activation_date`:
```php
$diff = strtotime($activation_date) - strtotime(gmdate('Y-m-d H:i:s'));
return (int) abs(round($diff / 86400));
```
- Returns negative values for past dates (normal case), then uses `abs()` to make positive
- Could mask data corruption if activation dates are somehow in the future
- The `abs()` call suggests the original formula might be backwards

### 2. `edac_parse_html_for_media()` - Overly Complex Regex
**Issue**: Uses unnecessarily complex regex pattern:
```php
'/<img[^>]*(?:^|[\s>])src=([\'"])(.*?)\1[^>]*>/i'
```
- The `(?:^|[\s>])` attempts to prevent `data-src` matching but adds complexity
- Could fail with malformed HTML in edge cases
- A simpler pattern might be more reliable

### 3. Environment-Dependent Functions
**Issues Found**:
- `edac_check_plugin_installed()` depends on `get_plugins()` which may not be available in all contexts
- `Helpers::is_domain_loopback()` relies on DNS functions that can fail in restricted environments
- Multiple functions require WordPress context that may not always be present

### 1. `Helpers::array_to_sql_safe_list()` - Proper SQL Escaping
- Correctly escapes malicious input: `"'; DROP TABLE"` becomes `"\\'; DROP TABLE"`
- Preserves data while making it safe (this is correct behavior)

### 2. `edac_generate_landmark_link()` - WordPress Standards Compliant
- Properly escapes HTML and URLs using WordPress functions
- Handles malicious input appropriately with `esc_html()`, `esc_url()`, etc.

## Recommendations:

1. **Review `edac_days_active()`** - Consider changing to `current_date - activation_date` for intuitive behavior
2. **Simplify HTML parsing regex** - The current pattern works but is more complex than needed
3. **Add environment checks** - Some functions assume WordPress context is always available

The tests revealed these quirks through edge case testing and environment compatibility issues encountered during development.

_Originally posted by @Copilot in https://github.com/equalizedigital/accessibility-checker/issues/1120#issuecomment-3139988488_
